### ISSUE TITLE
"Nullable object must have a value" after inserting module (randomly)

### ISSUE THREAD

**Issue by pascalmoser** [2sic/2sxc#139]
Sometimes when we insert a module on a page, choose a template and clicking `edit`, the following error message appears (only the current module instance is affected):

```
Fehler: Content ist zur Zeit nicht verfügbar. DotNetNuke.Services.Exceptions.ModuleLoadException: Nullable object must have a value. ---> 
 at .InvalidOperationException: Nullable object must have a value. at 
 at .Nullable`1.get_Value() at ToSic.SexyContent.SexyContent.GetViewDataSource(Int32 moduleId, Boolean showDrafts, Boolean includeEditingData, Nullable`1 overrideTemplateId)
 at ToSic.SexyContent.SexyContent.GetContentElements(Int32 ModuleID, String LanguageName, Nullable`1 ContentGroupID, Int32 PortalId, Boolean showDrafts)
 at ToSic.SexyContent.SexyControlEditBase.get_Elements() at ToSic.SexyContent.View.Page_PreRender(Object sender, EventArgs e)
 --- End of inner exception stack trace ---
```

After clearing the cache, everything works again. Does not occur often.

**Comment by iJungleboy** [comment]
Sounds like a very specific caching issue. 
Till more people report this, we'll put it on low-priority. If you can find further details (like when this usually happens) that would help.

**Comment by Tychodewaard** [comment]
Hi Daniel

Adding the GTM app is ok. Adding the GTM code in the app is ok. But it is a problem when I change the settings of the module to "display on all pages", than it gets cranky. Below the error from the event viewer.

A bien tot,
Tycho

DotNetNuke.Services.Exceptions.ModuleLoadException: Nullable object must have a value. ---> System.InvalidOperationException: Nullable object must have a value.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at ToSic.SexyContent.SexyContent.GetCompatibleTemplates(Int32 PortalID, Int32 ContentGroupID)
   at ToSic.SexyContent.SexyContent.GetAvailableTemplatesForSelector(ModuleInfo module)
   at ToSic.SexyContent.ViewApp.Page_PreRender(Object sender, EventArgs e)
   --- End of inner exception stack trace ---

**Comment by iJungleboy** [comment]
@Tychodewaard thanks. Can you specficy what versions of dnn and 2sxc you're using?

**Comment by Tychodewaard** [comment]
DNN 7.3.4 and 6.6.6 2SIC

**Comment by iJungleboy** [comment]
In a few hours we'll release 2sxc 7.0.1 which fixes an upgrade-issue. Could you check if the bug still appears?

**Comment by Tychodewaard** [comment]
A bug still exists cause I now I get "Error: Unkown The error has also been logged to the console."
Unfortunately, the error is not logged in the event viewer.

What I did find out:
The error only appears when the page is public for all users and the app is not. Or maybe I should say that the app has a problem when the permission is not set as inherited.

Because when the page is not public for all users (for instance admin and registered users) and the module inherits the permissions, it's not a problem.

So, in contradiction to what I assumed earlier: it has nothing to do with the setting 'display on all pages' because that works fine. 

Does this make any sense?
Most important thing: as the GTM app is not visible for non-admin I can inherited the settings, so the problem is not major.

Grtz
Tycho

**Comment by iJungleboy** [comment]
@Tychodewaard thanks for cornering the issue some more. 
So if I get it right, the issue appears when
a) a page is public
b) an module with an app is not

then apparently a security query seems to run into a null-exception, right?

**Comment by nokiko** [comment]
I now suddenly have this on one site whe i get this error every time i try to change the settings of a field

so if i want to change a string field to dropdown when i click string i get this error  the same goes for if I want to use an entity field

so at the moment I cant use entity fields anymore as i cant access them, no matter how much I clear cache or restart application

**Comment by iJungleboy** [comment]
@nokiko - you just sent me an e-mail saying you solved a problem. Was it this problem, or another one?

**Comment by nokiko** [comment]
well it was a workaround I got the above error but it was actually related to this issue ( only for me it did not show the content item page but generated the error )

https://github.com/2sic/2sxc/issues/258

workaround for me was that i just take the url that is in the error log, copy that remove the returnurl and then edit my entity setting

**Comment by raphael-m** [comment]
The original issue from @pascalmoser describes a bug that has been fixed with 2sxc 8.0.2. I'll close this now.
